src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperParseTreeRewriter.cs

```diff
@@ -15,16 +14,38 @@ namespace Microsoft.AspNet.Razor.Parser.TagHelpers.Internal
 {
     public class TagHelperParseTreeRewriter : ISyntaxTreeRewriter
     {
+        // From http://dev.w3.org/html5/spec/Overview.html#elements-0
+        private static readonly ISet<string> VoidElements = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
```

Comment:
`HashSet<string> VoidElements`. No reason to make all its be methods be virtual calls.